File: core/src/commonMain/kotlin/ru/itmo/json_forms/core/AdapterController.kt
package ru.itmo.json_forms.core

import ru.itmo.json_forms.core.document.Document

class AdapterController {

    fun startAdapterWith(document: Document, adapter: AbstractAdapter) {
        adapter.init()
        adapter.renderJson(document.root)
        adapter.onEnd()
    }

}

File: vscode-plugin/src/adapterController.ts
import * as jsonForms from 'json-forms-core';
import Document = jsonForms.ru.itmo.json_forms.core.document.Document;
import { HtmlAdapter } from './htmlAdapter';

export class AdapterController {

    public startAdapterWith(document: Document, adapter: HtmlAdapter) {
        adapter.init();
        adapter.renderJson(document.root);
        adapter.onEnd();
    }

}